### DISCUSSION TITLE
[Ideas] TOC 표시 유형 추가

### DISCUSSION THREAD

**Discussion by TetraTheta** [pronist/hELLO#67] [Ideas]
현재 TOC는 창 크기의 가로가 1280px 이상인 경우에만 화면 우측에 나타납니다. 이를 대략적인 그림으로 나타내면 이런 그림이 되겠지요.

![current](https://user-images.githubusercontent.com/37061801/160270242-a9361c91-8687-4907-9b47-33b7cb4a884e.png)

TOC의 표시 유형을 추가했으면 합니다.

![new](https://user-images.githubusercontent.com/37061801/160270269-66c78368-b402-4801-9901-0222f662dfef.png)

위와 같이, 본문 영역 맨 위에 TOC을 상시 표시하는 옵션이 추가되었으면 합니다.

여기에 추가적으로 적용 가능한 TOC 표시 유형 옵션을 제안합니다.

 - 화면 좌측 표시: 이 경우, 가로 1280px 이상인 경우에만 표시됩니다. 현재의 TOC 표시 설정과 일치합니다.
 - 본문 상단 표시: TOC를 가로 크기에 상관 없이 본문 상단에 표시합니다.
     - 모두 표시: 전체 TOC을 표시합니다. 이 경우 TOC이 길어져 스크롤을 내려야만 본문을 볼 수 있는 경우가 생길 수 있습니다. 따라서 다음 옵션이 존재합니다.
     - 접은 상태로 표시: TOC를 접은 상태로 표시합니다. 이 경우 TOC이 존재한다는 의미로 TOC이 '목차(혹은 Table of Content)' 한 줄로 표시되며, 이를 클릭해 전체 TOC를 '펼칠' 수 있습니다.
 - 없음: 가로 1280px 여부와 관계없이 TOC를 일절 표시하지 않습니다.

---

가능하다면 본문 내 `<script>` 태그를 통해 이러한 TOC 설정을 오버라이드 할 수 있으면 좋겠습니다.  
가령, 스킨 설정에서 TOC을 '없음'으로 설정하더라도 본문 상단에 HTML 입력 옵션을 통해 `<script>toc_display = 'topall';</script>`를 입력한다면, 해당 글에서는 TOC이 '본문 상단 표시 - 모두 표시'로 나타나는 것입니다.  

 - 화면 좌측 표시: `side`
 - 본문 상단 표시:
     - 모두 표시: `topall`
     - 접은 상태로 표시: `topfold`
 - 없음: `none`

다만 이 경우, TOC과 관련된 스크립트가 `<article>` 태그보다 아래에 위치해 있어야 정상적으로 작동한다는 점이 단점입니다.

**Comment by pronist** [comment]
@TetraTheta 아이디어 감사합니다. 본문 위에 TOC 를 표시하는 것은 댓글로 한 번 건의된 적이 있어서 이 부분은 처리가 될 예정이기는 했습니다. 구체적인 내용까지 제시해주신 부분에 대해서는 감사할 따름이군요. 제시해주신 부분이 날것 그대로 처리되지는 않겠지만 구현할 때 참고하도록 하겠습니다. :)
